devicetypes/smartthings/zigbee-metering-plug-power-consumption-report.src/zigbee-metering-plug-power-consumption-report.groovy

```diff
@@ -0,0 +1,185 @@
+                        map.unit = "W"
+                }
+                else if (it.value && it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_READING_INFO_SET) {
+                        log.debug "energy"
+                        map.name = "energy"
+                        map.value = zigbee.convertHexToInt(it.value)/getEnergyDiv()
+                        map.unit = "kWh"
+                        
+                        if (isDawonOutlet()) { 
+                            def currentEnergy = map.value
+                            def currentPowerConsumption = device.currentState("powerConsumption")?.value
+                            Map previousMap = currentPowerConsumption ? new groovy.json.JsonSlurper().parseText(currentPowerConsumption) : [:]
+                            def deltaEnergy = calculateDelta (currentEnergy, previousMap)
+                            Map reportMap = [:]
+                            reportMap["energy"] = currentEnergy
+                            reportMap["deltaEnergy"] = deltaEnergy 
+                            sendEvent("name": "powerConsumption", "value": reportMap.encodeAsJSON(), displayed: false)
+                        }
+                }
+
+                if (map) {
+                        result << createEvent(map)
+                }
+                log.debug "Parse returned $map"
+        }
+        return result
+    }
+}
+
+def off() {
+    def cmds = zigbee.off()
+    return cmds
+}
+
+def on() {
+    def cmds = zigbee.on()
+    return cmds
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return refresh()
+}
+
+def refresh() {
+    log.debug "refresh"
+    zigbee.onOffRefresh() +
+    zigbee.electricMeasurementPowerRefresh() +
+    zigbee.readAttribute(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET)
+}
+
+def configure() {
+    // this device will send instantaneous demand and current summation delivered every 1 minute
+    sendEvent(name: "checkInterval", value: 2 * 60 + 10 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+    log.debug "Configuring Reporting"
+    return refresh() +
+    	   zigbee.onOffConfig() +
+           zigbee.configureReporting(zigbee.SIMPLE_METERING_CLUSTER, ATTRIBUTE_READING_INFO_SET, DataType.UINT48, 1, 600, 1) +
+           zigbee.electricMeasurementPowerConfig(1, 600, 1) +
+           zigbee.simpleMeteringPowerConfig()
+}
+
+private int getPowerDiv() {
+    1
+}
+
+private int getEnergyDiv() {
+    if (isDawonOutlet()) {
+        1000
+    } else {
+        100
+    }
+}
+
+BigDecimal calculateDelta (BigDecimal currentEnergy, Map previousMap) {
+    if (previousMap?.'energy' == null) {
+        return 0;
+    }
+    BigDecimal lastAcumulated = BigDecimal.valueOf(previousMap ['energy']);
+    return currentEnergy.subtract(lastAcumulated).max(BigDecimal.ZERO);
+}
+
+private boolean isDawonOutlet() {
```

Comment:
this is true for all the devices on this DTH now, so the code should be refactored accordingly

Reply:
I'll erase it.